jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/oqua/ExportRepository.kt

```diff
@@ -0,0 +1,228 @@
+package org.wycliffeassociates.otter.jvm.workbookapp.oqua
+
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import org.wycliffeassociates.otter.common.data.workbook.Chapter
+import org.wycliffeassociates.otter.common.data.workbook.Workbook
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.PrintWriter
+import javax.inject.Inject
+
+/** TODO
+ * Implement Export
+ * Must be a function that takes a chapter and exports its saved results
+ *
+ * When in chapter view, save results and export the chapter
+ * When in project view, export each chapter
+ *
+ * Must be able to grab questions from chapters using the draftReviewRepo
+ * Probably will be an interface that gets injected into both project and chapter view model
+ *
+ * Maybe have it be a view model? Sounds wrong
+ * -- Have it be a Component
+ * -- Nope! Check out DraftReviewRepository. It has an injected Directory Provider
+ */
+
+class ExportRepository @Inject constructor (
+    private val draftReviewRepo: DraftReviewRepository,
+    private val questionsRepo: QuestionsRepository
+) {
+    private val disposables = CompositeDisposable()
+
+    fun exportChapter(workbook: Workbook, chapter: Chapter, directory: File, callback:(Workbook, Chapter, Boolean) -> Unit) {
+        lateinit var reviews: ChapterDraftReview
+        try {
+            reviews = draftReviewRepo.readDraftReviewFile(workbook, chapter)
+            writeReviewsToFile(
+                reviews,
+                directory,
+                packageCallback(callback, workbook, chapter)
+            )
+        } catch (_: FileNotFoundException) {
+            handleMissingReview(
+                workbook,
+                chapter,
+                directory,
+                packageCallback(callback, workbook, chapter)
+            )
+        }
+    }
+
+    private fun packageCallback(
```

Comment:
this method takes in a lambda, calls the lambda, and returns a lambda...

We may not be able to understand what was going on after a few weeks and come back reading this.